crates/chaintsn/src/context.rs

```diff
@@ -80,16 +81,16 @@ impl BlockHeaderContext for L2HeaderAndParent {
 /// Accessor for fetch and manipulate the state we're building on top of.
 ///
 /// This is supersceding the `StateCache` type.
-pub trait StateAccessor {
+pub trait StateAccessor<S = Chainstate> {
     /// Gets a ref to the state.
     ///
     /// This is a transitional accessor that we will deprecate and remove soon.
-    fn state_untracked(&self) -> &Chainstate;
+    fn state_untracked(&self) -> &S;
 
     /// Gets a mut ref to the state.
     ///
     /// This is a transitional accessor that we will deprecate and remove soon.
-    fn state_mut_untracked(&mut self) -> &mut Chainstate;
+    fn state_mut_untracked(&mut self) -> &mut S;
```

Comment:
Actually should be removing this function.

Reply:
This is being used at a bunch of places. Will remove it gradually.